ballista/rust/executor/src/executor_server.rs

```diff
@@ -310,25 +354,38 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskRunnerPool<T,
         Self { executor_server }
     }
 
-    // There are two loop(future) running separately in tokio runtime.
-    // First is for sending back task status to scheduler
-    // Second is for receiving task from scheduler and run
-    async fn start(
+    fn start(
         &self,
         mut rx_task: mpsc::Receiver<TaskDefinition>,
         mut rx_task_status: mpsc::Receiver<TaskStatus>,
+        graceful: &Graceful,
     ) {
         //1. loop for task status reporting
         let executor_server = self.executor_server.clone();
+        let mut tasks_status_shutdown = graceful.subscribe_for_shutdown();
+        // Not used directly.
+        let tasks_status_complete = graceful.shutdown_complete_tx.clone();
         tokio::spawn(async move {
             info!("Starting the task status reporter");
-            loop {
+
+            // Drop and notifies the receiver half that the shutdown is complete
```

Comment:
```suggestion
            // Capture `task_status_complete`. This will be dropped when this background task returns, indicating that shutdown is complete
```